src/notifications/diskspacenotifier.cpp

```diff
@@ -63,9 +63,24 @@ void DiskSpaceNotifier::handleDiskSpaceChange(const QString &path, int percentag
         QVariantHash hints;
         hints.insert(NotificationManager::HINT_CATEGORY, "x-nemo.system.diskspace");
         //% "Getting low with storage. Please check."
-        hints.insert(NotificationManager::HINT_PREVIEW_BODY, qtTrId("qtn_memu_memlow_notification_src"));
+        QString diskLowText = qtTrId("qtn_memu_memlow_notification_src");
+        hints.insert(NotificationManager::HINT_PREVIEW_BODY, diskLowText);
         // TODO go to some relevant place when clicking the notification
         notificationId = manager->Notify(qApp->applicationName(), 0, QString(), QString(), QString(), QStringList(), hints, -1);
+
+        bool publishPermanent = true;
+        NotificationList notifications = manager->GetNotifications(qApp->applicationName());
+        foreach (LipstickNotification* notification, notifications.notifications()) {
+            if (notification->category() == "x-nemo.system.diskspace.permanent") {
+                publishPermanent = false;
+                break;
+            }
+        }
+        if (publishPermanent) {
+            QVariantHash permanentHints;
+            permanentHints.insert(NotificationManager::HINT_CATEGORY, "x-nemo.system.diskspace.permanent");
+            manager->Notify(qApp->applicationName(), 0, QString(), diskLowText, QString(), QStringList(), permanentHints, -1);
```

Comment:
Only the "non-permanent" notifications are removed. The original disk space low notification cannot be used in the events view, because it's in special "system notification" category (indicated by "urgency == 2"). Making it a "non-system category" notification would change the behaviour in many other places. Also we need to use different icon for both.